src/webgpu/api/operation/command_buffer/copies.spec.ts

```diff
@@ -17,13 +18,217 @@ Test Plan:
   first copy from srcBuffer to a region of dstBuffer, then copy from another part of srcBuffer to
   another region of dstBuffer which have overlaps with the region of dstBuffer in the first copy
   and check the content of the whole dstBuffer to see the copies are done in correct order.
+2. CopyTextureToTexture
+* Validate the correctness of the copy by filling the srcTexture with testable data and any format that is
+  supported by WebGPU, doing CopyTextureToTexture() copy, and verifying the content of the whole dstTexture:
+  Copy {1 texel block, part of, the whole} srcTexture to the dstTexture {with, without} a non-zero valid
+  srcOffset that
+  - covers the whole dstTexture subresource
+  - covers the corners of the dstTexture
+  - doesn't cover any texels that are on the edge of the dstTexture
+  - covers the mipmap level > 0 
+  - covers {one, multiple} 2D texture array slices
 `;
 
 import { poptions, params } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/framework/util/util.js';
+import { kAllTextureFormatInfo, kRegularTextureFormats } from '../../../capability_info.js';
 import { GPUTest } from '../../../gpu_test.js';
+import { align } from '../../../util/math.js';
+import { getTexelDataRepresentation } from '../../../util/texture/texelData.js';
+
+class F extends GPUTest {
+  GetPhysicalSubresourceSize(
```

Comment:
This could probably moved to a helper in utils.

Reply:
And take a `GPUTextureDimension` because the depth will be mip-mapped for 3D textures too.